tests/test_templates.py

```diff
@@ -12,106 +12,307 @@
-                            random_layers_normal, random_layer_normal,
-                            cvqnn_layers_uniform, cvqnn_layer_uniform,
-                            cvqnn_layers_normal, cvqnn_layer_normal)
+from pennylane.templates.layers import Interferometer
+from pennylane.templates.layers import (CVNeuralNetLayers,
+                                        StronglyEntanglingLayers,
+                                        RandomLayers)
+from pennylane.templates.embeddings import (AmplitudeEmbedding,
+                                            BasisEmbedding,
+                                            AngleEmbedding,
+                                            SqueezingEmbedding,
+                                            DisplacementEmbedding)
+from pennylane.init import (strong_ent_layers_uniform,
+                            strong_ent_layers_normal,
+                            random_layers_uniform,
+                            random_layers_normal,
+                            cvqnn_layers_uniform,
+                            cvqnn_layers_normal)
 
+# When adding a new template,
+# extend the appropriate lists with the template function
+# and the desired inputs (i.e., features and/or weights) that need to be tested
 
-class TestParameterIntegration:
-    """ Integration tests for the parameter generation methods from pennylane.init
-    and pennylane.templates.layers."""
+fixture = "template, inpts"
 
-    @pytest.mark.parametrize('parfun', [cvqnn_layers_uniform, cvqnn_layers_normal])
-    def test_integration_cvqnn_layers(self, parfun, gaussian_device, n_subsystems, n_layers):
-        """Checks that pennylane.init.cvqnn_layers_uniform() integrates
-        with pennnylane.templates.layers.CVNeuralNetLayers()."""
+# Constant input fixtures for qubit templates
+qubit_func = [(StronglyEntanglingLayers, strong_ent_layers_uniform),
+              (StronglyEntanglingLayers, strong_ent_layers_normal),
+              (RandomLayers, random_layers_uniform),
+              (RandomLayers, random_layers_normal)]
 
-        p = parfun(n_layers=n_layers, n_wires=n_subsystems)
+# Constant input fixtures for continuous-variable templates
+cv_func = [(CVNeuralNetLayers, cvqnn_layers_uniform),
+           (CVNeuralNetLayers, cvqnn_layers_normal)]
 
-        @qml.qnode(gaussian_device)
-        def circuit(weights):
-            CVNeuralNetLayers(*weights, wires=range(n_subsystems))
+# Constant input fixtures for qubit templates
+qubit_const = [(RandomLayers, [[[0.53479316, 5.88709314], [2.21352321, 4.28468607]]]),
+               (AmplitudeEmbedding, [[1 / 2, 1 / 2, 1 / 2, 1 / 2]]),
+               (BasisEmbedding, [[1, 0]]),
+               (AngleEmbedding, [[1., 2.]])]
+
+# Constant input fixtures for continuous-variable templates
+cv_const = [(DisplacementEmbedding, [[1., 2.]]),
+            (SqueezingEmbedding, [[1., 2.]]),
+            (CVNeuralNetLayers, [[[2.33312851], [1.20670562]],
```

Comment:
How were these chosen?

Reply:
Some computer random, some Maria random ;) Any preference?

Reply:
Since it looks like they are just chosen to have the right type and shape, but we don't care about the actual values, nope